### DISCUSSION TITLE
Handle if (gltfRoot.scene < 0) better?

### DISCUSSION THREAD

**Discussion by andypoly** [atteneder/glTFast#450] [Ideas]
in GltfImport code it has the line:
`if (gltfRoot.scene < 0) return true;`

Now I understand this is correct gltf specification but the 2 issues are:
1. You get no warning if a model has no scene index - so you do not know what happened without debugging this bit of code
2. The 3ds max exporter does not seem to assign a root scene, so unless you force it to be 0 (when -1) you will not load a model. It may be the built in exporter is just a bit rubbish!

**Comment by atteneder** [comment]
Thanks for the feedback!

Yes, someone needs to tell the 3ds max exporter devs that not setting a scene is a bad default behavior (and bug imho).

I agree that `InstantiateMainScene` can be confusing. Solution ideas:

1. Log a warning to the console ( in DEBUG mode only)
2. Return `false` if no main scene was set
  - More explicit that "something went wrong"
  - No differentiation if instantiation failed in some way or no scene id was set
3. Return a new state enum, which can be one of:
  - Success
  - Error
  - NoSceneSet
  - ...

Imho 1 seems like an ad-hoc improvement while 3 is probably the cleanest solution.

Opinions?

**Reply by andypoly** [reply]
yes agreed

But for now I changed the code to set a -1 scene index to 0 because we have an artist using 3ds max!
This was only way to load them - maybe we will find a better exporter plugin.

So if you fancy adding an override to deal with bad exporters it could be an option?!

**Reply by atteneder** [reply]
Can you elaborate what you mean with  "adding an override"? How does it compare to the proposed 3 solutions?

**Reply by andypoly** [reply]
Sorry I meant on the case of scene index being invalid for 3ds exports. I had to change the code for this to fix the bad index.
It would be useful (if not accurate to the gltf format) to have a fixSceneIndex option!

Now of course you don't have to do this, but if 3ds is slow to fix their exporter it may be useful to people

**Reply by atteneder** [reply]
Well, you can use the `sceneId` property of the `GltfAsset` component or via custom code (as you already do). I see no reason to add another way.

**Reply by andypoly** [reply]
ah I did not notice the GltfAsset sceneId - fine thanks

**Comment by atteneder** [comment]
For now, I've added a warning (in Editor and development builds only) and extended the XML docs. Hope that avoids future confusion.
